### ISSUE TITLE
Doc Preflight "Manager driving property" validation.

### ISSUE THREAD

**Issue by elliotcmorris** [OpenAssetIO/OpenAssetIO#1223]
Noting this so I don't forget.

**tl:dr: Is the host putting something in preflight traitsdata a hard assertion that it intends not to allow the manager to drive those values ... answer : probably yes, and we need to doc it.**

With preflight, there's this idea that you provide the traits that you don't expect the manager to drive, (this relates to us needing to implement the awfully named 'kDerived' access, so hosts can know which specific traits these are when using management policy.)

The idea occurs that a host could provide a trait that the manager itself expects to be driving, and dosen't want to allow the host to move forward under the expectation that the host gets to decide this. Imagine a render node, and the location property is provided by the host in preflight, despite the fact that the manager knows that the render nodes _must_ stream their data to specific locations on the cluster.

Now, the manager can currently use the error callback to convey this, but it strikes me that we've never thought about this, and it isn't docced anywhere, whether this pattern is a recommendation or just a emergent effect of the ecosystem. We should think about it then doc something. 

# AC
- Doc for "manager driven" traits should include the points: 
  * a manager wishes to drive some/all of the properties of this trait.
  * if a property is set on the trait by the manager, then the host should use and publish this value.
  * hosts _may_ choose to ignore the value or not bother `resolve`ing for `kManagerDriven` at all, but must be aware that  `register` may fail.
- Update API docs for `kManagerDriven` enum value.
- Update `managementPolicy` Jupyter Notebook to highlight as a callout.

**Comment by elliotcmorris** [comment]
This is a doc issue, does `kManagerDriven` mean that the trait _must_ be manager driven. The answer is no, it can't.

**Comment by feltech** [comment]
I'm starting to think the old idea of a "dry run" access mode for `preflight` (perhaps other methods too) would help simplify a lot of introspection method design, or nasty "you just have to try and see" workflows.
